### PR TITLE
Get mime fallback

### PR DESCRIPTION
**Pull request by derkan**
## Description

With current `filesystem` middleware, we don't let user to set custom content-type for a new extension when called from [here](https://github.com/gofiber/fiber/blob/master/middleware/filesystem/filesystem.go#L179). Becuse fiber's [getMIME](https://github.com/gofiber/fiber/blob/b634ba0a58315308f488a4a5a920d46f095465fc/utils/http.go#L14) function uses statically defined mime types which cannot be changed. 

This is needed when serving SPA project from emdedded filesystem where custom contet-types are needed. (i.e.[sveltekit *.mjs file](https://github.com/sveltejs/kit/issues/9141))

### Fixes: 
Allows fallback to golang's [mime package](https://pkg.go.dev/mime) for content-type detection. Because, go's default mime package allows setting custom mime types.

## Type of change

Please delete options that are not relevant.

-  [x] Bug fix (non-breaking change which fixes an issue)
-  [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist:

- [ ] For new functionalities I follow the inspiration of the express js framework and built them similar in usage
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation - https://github.com/gofiber/docs for https://docs.gofiber.io/
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] If new dependencies exist, I have checked that they are really necessary and agreed with the maintainers/community (we want to have as few dependencies as possible)
- [x] I tried to make my code as fast as possible with as few allocations as possible
- [ ] For new code I have written benchmarks so that they can be analyzed and improved

### CODE REVIEW

**Comment by gaby** [comment]
@derkan Some small errors showing up in the tests/lint

**Comment by gaby** [comment]
@derkan Are the tests passing locally for you?

**Review by ReneWerner87 — CHANGES_REQUESTED** [review]
please check the failed tests

**Comment by derkan** [comment]
> @derkan Are the tests passing locally for you?

yes

**Comment by derkan** [comment]
problem is with go.1.16. 

For the following code:

```go
package main

import (
	"fmt"
	"mime"
)

func main() {
	fmt.Println("result:", mime.TypeByExtension(".text"))
}
```

- go.1.19 and go.1.20 this prints:

```
result: 
```

- go.1.16 prints:

```
text/plain; charset=utf-8
```

And because of this, [Test_Ctx_Accepts](https://github.com/gofiber/fiber/blob/master/ctx_test.go#L52) fails for go.1.16 at line:

```go
utils.AssertEqual(t, "json", c.Accepts("json", "text"))
```

File: utils/http.go [hunk 1]
```diff
@@ -5,6 +5,7 @@
 package utils
 
 import (
+	goMime "mime"
```
Comment (leonklingele) [inline utils/http.go:8]: Please don't use import-aliases for builtin packages. Better rename the variables below to something more or less meaningful.

File: utils/http_test.go [hunk 2]
```diff
@@ -23,6 +23,14 @@ func Test_GetMIME(t *testing.T) {
 
 	res = GetMIME("unknown")
 	AssertEqual(t, MIMEOctetStream, res)
+
+	if err := mime.AddExtensionType(".mjs", "application/javascript"); err == nil {
+		res = GetMIME(".mjs")
+		AssertEqual(t, "application/javascript", res)
+	} else {
+		t.Logf("failed to add extension type, err: %v", err)
```
Comment (leonklingele) [inline utils/http_test.go:31]: This must also never fail -- use `AssertEqual(t, nil, err)` here instead.

**Comment by ReneWerner87** [comment]
https://expressjs.com/en/5x/api.html#req.accepts
image

we need a solution for this case
@derkan  any ideas?

File: utils/http.go (hunk 1 above)
Comment (derkan) [reply utils/http.go]: @leonklingele thanks for reviewing. done

File: utils/http_test.go (hunk 2 above)
Comment (derkan) [reply utils/http_test.go]: @leonklingele thanks for reviewing. done

**Comment by derkan** [comment]
@ReneWerner87 a wrapper function(for making variadic - array type for mime names) around [mine.ParseMediaType(...)](https://github.com/golang/go/blob/b94dc384cabf75e7e8703265cd80f5324f84b642/src/mime/mediatype.go#L139) can be enough? [Playground](https://goplay.tools/snippet/oqpnDzP7IJr)

File: utils/http_test.go [hunk 3]
```diff
@@ -24,12 +24,12 @@ func Test_GetMIME(t *testing.T) {
 	res = GetMIME("unknown")
 	AssertEqual(t, MIMEOctetStream, res)
 
-	if err := mime.AddExtensionType(".mjs", "application/javascript"); err == nil {
-		res = GetMIME(".mjs")
+	err := mime.AddExtensionType(".mjs", "application/javascript")
+	if err == nil {
+		res = GetMIME(".text")
```
Comment (ReneWerner87) [inline utils/http_test.go]: why did you change the test case?

**Comment by ReneWerner87** [comment]
@gaby @leonklingele @efectn any ideas or opinions ?

**Comment by gaby** [comment]
> @gaby @leonklingele @efectn any ideas or opinions ?

It's a nice feature to add. I'm confused as to why the tests dont' pass for go 1.16 though. 

May be worth making sure our unit-tests match the expected results as shown in https://expressjs.com/en/5x/api.html#req.accepts

**Review by leonklingele — APPROVED** [review]
lgtm! 🚀

**Comment by gaby** [comment]
@derkan Can you update your branch with the latest changes in master. Trying to see if tests will pass.

**Comment by gaby** [comment]
CI is passing now!

**Comment by leonklingele** [comment]
It’s just flaky.

**Review by ReneWerner87 — APPROVED** [review]

**Comment by ReneWerner87** [comment]
i have made a sync with the upstream, otherwise you don't see the real changes because of all the copies(144 files) of the changes from the masters
